src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrBackflower.cs

```diff
@@ -450,24 +452,14 @@ private async Task RecreatePreviousFlowAndApplyBuild(
             line => line.Contains(VersionDetailsParser.SourceElementName) && line.Contains(lastFlow.SourceSha),
             lastFlow.RepoSha);
 
-        // Find the ID of the last VMR build that flowed into the repo
-        VersionDetails versionDetails = _versionDetailsParser.ParseVersionDetailsFile(targetRepo.Path / VersionFiles.VersionDetailsXml);
-
         // checkout the previous repo sha so we can get the last last flow
         await targetRepo.CheckoutAsync(previousRepoSha);
         await targetRepo.CreateBranchAsync(headBranch, overwriteExistingBranch: true);
         (Codeflow lastLastFlow, _, _) = await GetLastFlowsAsync(mapping, targetRepo, currentIsBackflow: true);
 
-        Build previouslyAppliedVmrBuild;
-        if (versionDetails.Source?.BarId != null)
-        {
-            previouslyAppliedVmrBuild = await _barClient.GetBuildAsync(versionDetails.Source.BarId.Value);
-        }
-        else
+        Build previouslyAppliedVmrBuild = new(-1, DateTimeOffset.Now, 0, false, false, lastLastFlow.SourceSha, [], [], [], [])
         {
-            // If we don't find the previously applied build, there probably wasn't one.
-            // In this case, we won't update assets, but that's ok because they'll get overwritten by the new build anyway
-            previouslyAppliedVmrBuild = new(-1, DateTimeOffset.Now, 0, false, false, lastLastFlow.VmrSha, [], [], [], []);
+            GitHubRepository = repoGitHubUri
```

Comment:
The VmrForwardFlower also sets the AzDO URI here. Should we have that too?

Reply:
Functionality wise, it doesn't matter, but for consistency I think it might be worth it.
The reason it doesn't technically matter is because we get this uri from `build.GetRepository()`, which will pass a GH repo if it exists, otherwise an AzDo repo, and we use the same method later to get the remote uri. But it's confusing like this